packages/location-management/src/components/LocationUnitGroupAddEdit/tests/Form.test.tsx

```diff
@@ -0,0 +1,227 @@
+  });
+
+  it('renders without crashing', () => {
+    const wrapper = mount(
+      <Provider store={store}>
+        <Router history={history}>
+          <Form />
+        </Router>
+      </Provider>
+    );
+    expect(wrapper.props()).toMatchSnapshot();
+  });
+
+  it('tests cancel button', () => {
+    const mockBack = jest.fn();
+    history.goBack = mockBack;
+
+    const wrapper = mount(
+      <Provider store={store}>
+        <Router history={history}>
+          <Form />
+        </Router>
+      </Provider>
+    );
+
+    wrapper.find('button#cancel').simulate('click');
+
+    // click go back
+    expect(wrapper.find('button').first().text()).toMatchInlineSnapshot(`"Save"`);
+    wrapper.find('button').first().simulate('click');
+
+    // click go back
+    expect(wrapper.find('button').last().text()).toMatchInlineSnapshot(`"Cancel"`);
+    wrapper.find('button').last().simulate('click');
+
+    expect(mockBack).toHaveBeenCalled();
+  });
+
+  it('tests Create New Payload', async () => {
+    const wrapper = mount(
+      <Provider store={store}>
+        <Router history={history}>
+          <Form />
+        </Router>
+      </Provider>
+    );
+
+    // with values test
+    wrapper
+      .find('input[name="name"]')
+      .simulate('change', { target: { name: 'name', value: 'Name213' } });
+    wrapper.update();
+    wrapper
+      .find('textarea[name="description"]')
+      .simulate('change', { target: { name: 'description', value: 'this is description' } });
+    wrapper.update();
+    wrapper.simulate('submit');
+
+    await act(async () => {
+      await flushPromises();
+      wrapper.update();
+    });
+
+    wrapper.unmount();
+  });
+
+  it('tests Update Payload', async () => {
+    fetch.once(JSON.stringify(fixtures.sampleLocationUnitGroupPayload));
+    const wrapper = mount(
+      <MemoryRouter initialEntries={[`/1`]}>
+        <Provider store={store}>
+          <Route path={'/:id'} component={() => <Form id="1" />} />
+        </Provider>
+      </MemoryRouter>
+    );
+
+    await act(async () => {
+      await flushPromises();
+      wrapper.update();
+    });
+
+    expect(fetch.mock.calls[0]).toEqual([
+      'https://opensrp-stage.smartregister.org/opensrp/rest/location-tag/1',
+      {
+        headers: {
+          accept: 'application/json',
+          authorization: 'Bearer null',
+          'content-type': 'application/json;charset=UTF-8',
+        },
+        method: 'GET',
+      },
+    ]);
+
+    wrapper
+      .find('input[name="name"]')
+      .simulate('change', { target: { name: 'name', value: 'Name213' } });
+    wrapper.update();
+    wrapper
+      .find('textarea[name="description"]')
+      .simulate('change', { target: { name: 'description', value: 'this is description' } });
+    wrapper.update();
+    wrapper.simulate('submit');
+
+    await act(async () => {
+      await flushPromises();
+      wrapper.update();
+    });
+
+    expect(fetch.mock.calls[0]).toEqual([
```

Comment:
Do we need t repeat this assertion since it's already implemented above

Reply:
it may look like same but they have different implementation for updating  and creating also in case of editing the data is fetched from server and pushed to form as initial state, also for updating we have different api call and endpoint